packages/mds-types/index.ts

```diff
@@ -38,12 +38,244 @@ export const RULE_UNIT_MAP = {
   hours: 60 * 60
 }
 
+// Event Streaming
+export interface InboundEvent {
+  device_id: string
+  provider_id: string
+  event_type: string
+  event_type_reason: string
+  telemetry: {
+    device_id: string
+    timestamp: number
+    gps: { lat: number; lng: number; altitude: number; heading: number; speed: number; accuracy: number }
+    charge: number
+    provider_id: string
+  }
+  timestamp: number
+  trip_id: string
+  recorded: number
+  telemetry_timestamp: number
+  service_area_id: string
+  id: number
+}
+
+export interface InboundTelemetry {
+  device_id: string
+  provider_id: string
+  timestamp: number
+  charge: number
+  gps: { lat: number; lng: number; altitude: number; heading: number; speed: number; accuracy: number }
+  recorded: number
+  id: number
+}
+
+export const CE_TYPES = Enum('mds.event', 'mds.telemetry')
+export type CE_TYPE = keyof typeof CE_TYPES
+
+export interface StateEntry {
+  vehicle_type: VEHICLE_TYPE
+  type: string
+  timestamp: Timestamp
+  device_id: UUID
+  provider_id: UUID
+  recorded: Timestamp
+  annotation_version: number
+  annotation?: AnnotationData | null
```

Comment:
Is the type union with `null` really necessary, since these are optional properties?

Reply:
`Optional` properties (denoted with the `?` modifier), essentially just tack on a union with `undefined`. So if you had `S: T` then `S?: T` is equivalent to `S: T | undefined`, this doesn't cover the case where you set the property to `null` because in JS-land `undefined !== null`. Totally worth seeing if it's necessary to even set these to null though wherever structure is being initialized, but removing that from the type annotation at the moment would be a breaking change.